### PR TITLE
Fixed editions table on books pages contains link to current page

### LINKED ISSUE #6040: Editions table on book pages contains link to the current page
One primary function of the editions table is to present other editions of the book that a patron is currently viewing.  Today, the first item in the editions table references the page that the table is on (more clearly: clicking on the first link in the table reloads the current book page).  Aside from a slight color difference, there is nothing that indicates that the first item in the table is the book currently being viewed by the patron.  Removing this item should make it slightly easier for patrons to quickly find other, more suitable, editions.

### Relevant url?

https://openlibrary.org/books/OL7824010M/Where_the_Red_Fern_Grows or any other book page

### Steps to Reproduce

1. Go to a book page.
2. Click the link in the first item of the editions table.

* Actual: The page reloads, leaving me somewhat confused...
* Expected: A different book page is loaded.

### Details

- **Logged in (Y/N)?** N
- **Browser type/version?** N/A
- **Operating system?** N/A
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

When table rows are being rendered, prevent the row that references the current page from being rendered.

### Related files

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/work/editions_datatable.html

### Stakeholders

@Hitansh-Shah

### PR DESCRIPTION
**Pull request by Hitansh-Shah**
Signed-off-by: Hitansh Shah <shah.hitanshsanjay.mat20@itbhu.ac.in>

Closes #6040 

On books page it removes the link of the current page from editions table

### Technical

The changes were made solely in [editions_datatable.html](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/work/editions_datatable.html). I changed the conditional statements inside the for loop for rendering template. I also declared a new variable `render_first_flag` with default value of `1` which decides when to render with `render_first=True` argument in `render_template` function. The variable helps to differentiate between the case where the already displayed edition appears in first iteration of for loop and the case where it does not.

### Testing

1. Go to https://openlibrary.org/books/OL7824010M/Where_the_Red_Fern_Grows or any other book (with multiple editions) page.
2. Check that the editions table does not contain the editions which is already displayed

### Screenshot

**Before:** 
image

**After:**
image

### CODE REVIEW

**Comment by Hitansh-Shah** [comment]
@jimchamp Please review the PR

File: openlibrary/templates/type/work/editions_datatable.html [hunk 1]
```diff
@@ -14,18 +14,20 @@
...
+            $if book.key != edition.key:
+              $if render_first_flag == 1:
+                <tr class="highlight">
+                  $:render_template("books/edition-sort", book, edition_sort_key, render_first=True)
+                </tr>
+                $ render_first_flag = 0
+              $else:
+                <tr>
+                  $:render_template("books/edition-sort", book, edition_sort_key)
+                </tr>
```
Comment (jimchamp) [inline openlibrary/templates/type/work/editions_datatable.html:30]:
```suggestion
            $if book.key != edition.key:
              <tr>
                $:render_template("books/edition-sort", book, edition_sort_key)
              </tr>
```
No need to render the row corresponding to the page that is being viewed.

File: openlibrary/templates/type/work/editions_datatable.html [hunk 2]
```diff
@@ -14,18 +14,20 @@
         $ edition_list_start = time()
         $ editions = editions or work.get_sorted_editions()
         $ index_padding = len(str(len(editions)))
+        $ render_first_flag = 1
```
Comment (jimchamp) [inline openlibrary/templates/type/work/editions_datatable.html:17]:
```suggestion
```

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @Hitansh-Shah!

I may have documented the issue incorrectly.  We don't want to display the selected item in the table at all, as it references the page that the user is currently viewing (i.e. clicking on the link in this row will reload the page).

**Comment by Hitansh-Shah** [comment]
@jimchamp Thanks for reviewing. So I was a little confused about `render_first=True` argument but I guess that is not needed at all now. I will update the changes. Thanks😄

**Comment by Hitansh-Shah** [comment]
@jimchamp  I have made the changes you suggested. Please review it :)

**Comment by jimchamp** [comment]
Thanks @Hitansh-Shah!  Unfortunately, I've led you astray.  It looks like we don't have consensus on removing the first editions table entry --- I thought that we did...

Closing this for now, but we may decide to merge it after some more internal discussion about the editions table.

**Pull request closed without merging**